src/client/webpack.config.js

```diff
@@ -43,7 +42,12 @@ const webpackConfig = {
     ]),
     new webpack.optimize.CommonsChunkPlugin(/* chunkName= */'vendor', /* filename= */'vendor.js', function(module){
       return module.resource && module.resource.indexOf('node_modules') !== -1;
-    })
+    }),
+    new webpack.DefinePlugin({
+     __CONFIG__: args.endpoint
```

Comment:
Just from my perspective: I'm unfamiliar with webpack. In the bootstrapper I see this `__CONFIG__` thing and a search takes me to webpack config which defines it from arguments. Doesn't seem that convoluted.

Comparatively the setting of `isProductionMode` seems more complicated.

And also top result for "webpack environment variables" is this: http://stackoverflow.com/a/30061249